src/background/redirects.spec.js

```diff
@@ -20,61 +20,85 @@ describe("Fandom redirect", () => {
-        {
-            url: "https://www.google.com/search?q=poe+wiki+faster+attacks",
-            expected: "https://www.google.com/search?q=site:poewiki.net+faster+attacks",
-        },
-        {
-            url: "https://www.google.com/search?q=poe++wiki+faster+attacks",
-            expected: "https://www.google.com/search?q=site:poewiki.net+faster+attacks",
-        },
-        {
-            url: "https://www.google.com/search?q=poe+faster+attacks+wiki",
-            expected: "https://www.google.com/search?q=site:poewiki.net+faster+attacks",
-        },
-        {
-            url: "https://www.google.com/search?q=faster+poe+attacks+wiki",
-            expected: "https://www.google.com/search?q=site:poewiki.net+faster+attacks",
-        },
-        {
-            url: "https://www.google.com/search?q=faster+attacks+poe+wiki",
-            expected: "https://www.google.com/search?q=site:poewiki.net+faster+attacks",
-        },
-        {
-            url: "https://www.google.com/search?q=faster+attacks+poewiki",
-            expected: "https://www.google.com/search?q=site:poewiki.net+faster+attacks",
-        },
-        {
-            url: "https://www.google.com/search?q=faster+poewiki+attacks",
-            expected: "https://www.google.com/search?q=site:poewiki.net+faster+attacks",
-        },
-        // Making sure words that happen to contain "poe", "wiki" or "poewiki" do not get filtered
-        {
-            url: "https://www.google.com/search?q=poe+apoe+poea+apoea+awiki+wikia+awikia+wiki+apoewiki+poewikia+apoewikia+poeawiki",
-            expected: "https://www.google.com/search?q=site:poewiki.net+apoe+poea+apoea+awiki+wikia+awikia+apoewiki+poewikia+apoewikia+poeawiki",
-        },
-    ])("Given $url, redirect to $expected", ({ url, expected }) => {
-        const actual = redirectFromGoogle({ url })
-        expect(actual.redirectUrl).toBe(expected)
-    })
+describe("Test queries against all Search Engines redirect", () => {
+    //Setup different Search Engine urls to test against
+    const googleUrls = [
+        "https://www.google.com/search?",
+        "https://google.com/search?",
+    ]
+
+    const duckduckgoUrls = [
+        "https://duckduckgo.com/?",
+        "https://www.duckduckgo.com/?"
+    ]
+
+    //The base redirect result URL to expect from each search engine
+    const redirectHost = new Map()
+    redirectHost.set("google", "https://www.google.com/search?q=site:poewiki.net+")
+    redirectHost.set("duckduckgo", "https://www.duckduckgo.com/?q=site:poewiki.net+")
+
+    //Run the tests for a given search engine
+    runTestQueries(googleUrls, "google")
+    runTestQueries(duckduckgoUrls, "duckduckgo")
+
+    function runTestQueries(urlCollection, searchEngine) {
+        urlCollection.forEach(urlHost =>
+            it.each([
+                {
+                    url: urlHost + "q=poe+faster+attacks",
+                    expected: redirectHost.get(searchEngine) + "faster+attacks"
```

Comment:
nit - minor duplication removal/optimization:
```suggestion
                    expected: redirectBaseUrl + "faster+attacks"
```

Under line 44
```js
const redirectBaseUrl = redirectBaseUrls[searchEngine]
```